Packs/AggregatedScripts/Scripts/QuarantineFile/QuarantineFile.py

```diff
@@ -0,0 +1,1144 @@
+
+    def _extract_quarantine_status_from_context(self, status_context: list) -> dict:
+        """
+        Extracts the core data dictionary from a command's nested entry context.
+
+        Args:
+            status_context (list): The entry context list from a command response.
+
+        Returns:
+            dict: The inner dictionary containing the status data, or an empty dict if not found.
+        """
+        demisto.debug(f"[{self.brand} Handler] Extracting status data from context: {status_context}")
+        if not status_context or not isinstance(status_context[0], dict):
+            return {}
+        # The context is often a list with one dict, where the key is the command name
+        # and the value is the actual data dictionary.
+        return list(status_context[0].values())[0]
+
+    def _process_final_endpoint_status(self, endpoint_result: dict, job_data: dict) -> QuarantineResult:
+        """
+        Processes the final result for a single endpoint from a completed polling job.
+
+        If the initial quarantine action was successful, this method makes a second,
+        separate call to 'get-quarantine-status' to get the true final result.
+
+        Args:
+            endpoint_result (dict): The result object for a single endpoint from the polling command's 'Contents'.
+            job_data (dict): The original job object containing metadata and finalize_args.
+
+        Returns:
+            QuarantineResult: A structured result object for the endpoint.
+        """
+        endpoint_id = endpoint_result.get("endpoint_id")
+        demisto.debug(f"[{self.brand} Handler] Processing final status for endpoint {endpoint_id}.")
+
+        if endpoint_result.get("status") == "COMPLETED_SUCCESSFULLY":
+            status_cmd = Command(
+                name=f"{self.command_prefix}-get-quarantine-status",
+                args={
+                    "endpoint_id": endpoint_id,
+                    "file_hash": job_data.get("finalize_args", {}).get("file_hash"),
+                    "file_path": job_data.get("finalize_args", {}).get("file_path"),
+                },
+                brand=self.brand,
+            )
+            raw_response, verbose_res = status_cmd.execute()
+            if self.orchestrator.verbose:
+                self.orchestrator.verbose_results.extend(verbose_res)
+
+            status_context = Command.parse_entry_context(raw_response)
+            quarantine_status_data = self._extract_quarantine_status_from_context(status_context)
+            quarantine_status = quarantine_status_data.get("status")
+            message = (
+                QuarantineResult.Messages.SUCCESS
+                if quarantine_status
+                else QuarantineResult.Messages.FAILED_WITH_REASON.format(reason=quarantine_status_data.get("error_description"))
+            )
+            status = QuarantineResult.Statuses.SUCCESS if quarantine_status else QuarantineResult.Statuses.FAILED
```

Comment:
maybe lets set the status once based on the quarantine_status

Reply:
please ellaborate

Reply:
nvm, my bad.